### PR TITLE
done task 3

### PR DESCRIPTION
**Pull request by demasha0908**
https://demasha0908.github.io/fe-react-2024/

- [x]  Mentor is added to pr
- [x]  AC copied to pr description
- [x]  Link to deployed version is added to pr
- [x]  The ProductsList component correctly receives product data as props.
- [x]  ProductCard component created and renders a single product.
- [x]  Header and Footer elements are present.
- [x]  ProductList displays right on mobile devices

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,14 +1,27 @@
...
 function App() {
+    const [isShowAbout, setIsShowAbout] = useState(true);
+
+    const toggleAbout = () => {
+        setIsShowAbout(true);
+    };
+
+    const toggleProducts = () => {
+        setIsShowAbout(false);
+    };
```
Comment (vladyslav-yermolin-moc) [inline src/App.tsx:19]: You can combine these two similar functions to one

File: src/components/product/Product.module.css [hunk 2]
```diff
@@ -0,0 +1,94 @@
+.product__card {
+  padding: 20px 15px 15px;
+  border: 1px solid #CCC;
+  border-radius: 20px;
+  margin: 30px 58px 0;
```
Comment (vladyslav-yermolin-moc) [inline src/components/product/Product.module.css:5]: Please correct the gaps 
image

File: src/components/product/Product.module.css [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+
+@media (min-width: 960px) {
+
+  .product__list {
+    grid-template-columns: repeat(3, 1fr);
+  }
+}
+
```
Comment (vladyslav-yermolin-moc) [inline src/components/product/Product.module.css:79]: Looks like you can replace these media queries by [auto-fill](https://developer.mozilla.org/en-US/docs/Web/CSS/repeat#auto-fill) prop

File: src/components/product/Product.module.css (hunk 2 above)
Comment (vladyslav-yermolin-moc) [reply src/components/product/Product.module.css:5]: If you use grid layout, the margins for its children are not needed. You can control inner indentation of the children by the grid parent `gap` prop.
Now your layout is broken 
image

File: src/components/header/Header.component.tsx [hunk 4]
```diff
@@ -9,52 +9,62 @@ import LightThemeBtn from '@/assets/sun.svg';
...
-            </nav>
+                    <nav className={styles.header__nav}>
+                        <ul className={styles.header__navwrapper}>
+                            <li>
+                                <button className={styles.header__linkactive} onClick={toggleAbout}>
+                                    About
+                                </button>
+                            </li>
+                            <li>
+                                <button className={styles.header__link} onClick={toggleProducts}>
```
Comment (vladyslav-yermolin-moc) [inline src/components/header/Header.component.tsx:40]: Please add the logic to change the active state of nav menu 
image

**Review by vladyslav-yermolin-moc — CHANGES_REQUESTED** [review]
Hey, @demasha0908, thanks for your work, and sorry for the long delay 🙏

**Comment by vladyslav-yermolin-moc** [comment]
@demasha0908, plis https://github.com/MastersAcademy/fe-react-2024/wiki/%5BFlow%5D-How-to-send-homework-for-review-and-get-task-done#how-to-fix-the-comments-of-your-mentor
>13. Go to the "Files changed" tab and check, are all your fixes in place. If so, add the comment "fixed" to every mentor's comment. **Please don't resolve the comments, the mentor will do it on his own!**

File: src/components/header/Header.component.tsx (hunk 4 above)
Comment (demasha0908) [reply src/components/header/Header.component.tsx]: fixed

File: src/components/product/Product.module.css (hunk 3 above)
Comment (demasha0908) [reply src/components/product/Product.module.css]: fixed

File: src/components/product/Product.module.css (hunk 2 above)
Comment (demasha0908) [reply src/components/product/Product.module.css]: fixed

File: src/App.tsx (hunk 1 above)
Comment (demasha0908) [reply src/App.tsx]: fixed

**Review by vladyslav-yermolin-moc — APPROVED** [review]

**Comment by demasha0908** [comment]
> @demasha0908, plis https://github.com/MastersAcademy/fe-react-2024/wiki/%5BFlow%5D-How-to-send-homework-for-review-and-get-task-done#how-to-fix-the-comments-of-your-mentor
> 
> > 13. Go to the "Files changed" tab and check, are all your fixes in place. If so, add the comment "fixed" to every mentor's comment. **Please don't resolve the comments, the mentor will do it on his own!**

oh, so sorry, didn`t see it

**Comment by vladyslav-yermolin-moc** [comment]
Good work 👍

**Pull request merged**